### PR TITLE
Merge from dotnet/runtime

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
@MichalStrehovsky @davidwrighton Could you please take a look at the MethodScope related build breaks? I am not sure what's the right way to fix them.

**Comment by MichalStrehovsky** [comment]
I'll take a look tomorrow. I was hoping this wouldn't affect us and I wouldn't need to understand what the split was trying to achieve.

**Comment by MichalStrehovsky** [comment]
The fix might be just along the lines of `using MethodILScope = Internal.TypeSystem.IL.MethodIL` at the top of the file but I don't have a clean enlistment to try out quckly. (Have been waiting for 5 months for a new laptop, postponing a Windows reinstall - which is the only way to get extra disk space on my primary work machine at this point).

File: src/coreclr/tools/aot/ILCompiler.Compiler/ILCompiler.Compiler.csproj [hunk 1]
```diff
@@ -448,6 +448,12 @@
     <Compile Include="..\..\Common\JitInterface\CorInfoInstructionSet.cs">
       <Link>JitInterface\CorInfoInstructionSet.cs</Link>
     </Compile>
+    <Compile Include="..\..\Common\JitInterface\CorInfoHelpFunc.cs">
+      <Link>JitInterface\CorInfoHelpFunc.cs</Link>
+    </Compile>
+    <Compile Include="..\..\Common\JitInterface\CorInfoTypes.cs">
```
Comment (MichalStrehovsky) [inline src/coreclr/tools/aot/ILCompiler.Compiler/ILCompiler.Compiler.csproj:454]: Sight, I thought we already included this file when I wrote https://github.com/dotnet/runtime/pull/53567#discussion_r648939767.

**Comment by jkotas** [comment]
@MichalStrehovsky Thank you!

**Pull request merged**